core/src/main/java/org/apache/calcite/sql/SqlGroupedWindowFunction.java

```diff
@@ -128,13 +128,6 @@ public SqlGroupedWindowFunction auxiliary(String name, SqlKind kind) {
     // make the method abstract.
     return call.getOperandMonotonicity(0).unstrict();
   }
-
-  @Override public String getName() {
-    // Always rename the name of the SqlKind. The tumble operator is called
-    // "$TUMBLE", so that it does not clash with a user-defined table function
-    // "TUMBLE", but we want the name to be "TUMBLE".
-    return getKind().name();
-  }
```

Comment:
Did you notice what the comment wants to state ? It says that although it's operator name is "$TUMBLE" but we still want to rename it to "TUMBLE" because this name is what we use in the SQL.

Reply:
Yes Danny. That was the old design to return "TUMBLE" as operator name.

It turns out the old design didn't fully solve function overloading problem (grouped TUMBLE and table value function TUMBLE). I discussed it with Julian in [1], and Julian recommended to return "$TUMBLE" directly as the function name. Please check the last two comments of [1].


[1]: https://jira.apache.org/jira/browse/CALCITE-3382

Reply:
Hi Danny! Do you have further comment?